### PR TITLE
[SPARK-25132][SQL][FOLLOWUP][DOC] Add migration doc for case-insensitive field resolution when reading from Parquet

### PR DESCRIPTION
**Pull request by seancxmao**
## What changes were proposed in this pull request?
#22148 introduces a behavior change. According to discussion at #22184, this PR updates migration guide when upgrade from Spark 2.3 to 2.4.

## How was this patch tested?
N/A

### CODE REVIEW

**Comment by seancxmao** [comment]
@HyukjinKwon Would you please kindly take a look at this when you have time?

File: docs/sql-migration-guide-upgrade.md [hunk 1]
```diff
@@ -141,6 +141,8 @@ displayTitle: Spark SQL Upgrading Guide
...
 
+  - In version 2.3 and earlier, when reading from a Parquet data source table, Spark always returns null for any column whose column names in Hive metastore schema and Parquet schema are in different letter cases, no matter whether `spark.sql.caseSensitive` is set to true or false. Since 2.4, when `spark.sql.caseSensitive` is set to false, Spark does case insensitive column name resolution between Hive metastore schema and Parquet schema, so even column names are in different letter cases, Spark returns corresponding column values. An exception is thrown if there is ambiguity, i.e. more than one Parquet column is matched. This change also applies to Parquet Hive tables when `spark.sql.hive.convertMetastoreParquet` is set to true.
```
Comment (dongjoon-hyun) [inline docs/sql-migration-guide-upgrade.md]: Hi, @seancxmao . Maybe, the followings?
```
- `spark.sql.caseSensitive` is set to true or false
+ `spark.sql.caseSensitive` is set to `true` or `false`
```
```
- `spark.sql.caseSensitive` is set to false
+ `spark.sql.caseSensitive` is set to `false`
```
```
- `spark.sql.hive.convertMetastoreParquet` is set to true
+ `spark.sql.hive.convertMetastoreParquet` is set to `true`
```

**Comment by dongjoon-hyun** [comment]
Thank you for adding this to the migration doc.
cc @gatorsmile .

File: docs/sql-migration-guide-upgrade.md (hunk 1 above)
Comment (seancxmao) [reply docs/sql-migration-guide-upgrade.md]: @dongjoon-hyun Good suggestions. I have fixed them with a new commit.

**Comment by dongjoon-hyun** [comment]
Merged to master and branch-2.4.

**Comment by dongjoon-hyun** [comment]
Thanks, @seancxmao .

**Comment by seancxmao** [comment]
Thank you! @dongjoon-hyun

**Pull request closed without merging**